src/test/java/org/jenkinsci/plugins/credentialsbinding/masking/AlmquistShellSecretPatternFactoryTest.java

```diff
@@ -0,0 +1,140 @@
+    public static final @DataPoint String FULL_ASCII = "!\"#$%&'()*+,-./ 0123456789:;<=>? @ABCDEFGHIJKLMNO PQRSTUVWXYZ[\\]^_ `abcdefghijklmno pqrstuvwxyz{|}~";
+
+    @DataPoints
+    public static List<String> generatePasswords() {
+        Random random = new Random(100);
+        List<String> passwords = new ArrayList<>(10);
+        for (int i = 0; i < 10; i++) {
+            int length = random.nextInt(24) + 8;
+            StringBuilder sb = new StringBuilder(length);
+            for (int j = 0; j < length; j++) {
+                // choose a (printable) character in the closed range [' ', '~']
+                // 0x7f is DEL, 0x7e is ~, and space is the first printable ASCII character
+                char next = (char) (' ' + random.nextInt('\u007f' - ' '));
+                sb.append(next);
+            }
+            passwords.add(sb.toString());
+        }
+        return passwords;
+    }
+
+    @ClassRule public static JenkinsRule j = new JenkinsRule();
+
+    private WorkflowJob project;
+    private String credentialsId;
+
+    @BeforeClass
+    public static void assumeBash() {
+        // ash = Almquist shell, default one used in Alpine
+        assumeThat("ash", is(executable()));
+        // due to https://github.com/jenkinsci/durable-task-plugin/blob/e75123eda986f20a390d92cc892c3d206e60aefb/src/main/java/org/jenkinsci/plugins/durabletask/BourneShellScript.java#L149
+        // on Windows
+        assumeThat("nohup", is(executable()));
+    }
+
+    @Before
+    public void setUp() throws IOException {
+        j.jenkins.getDescriptorByType(Shell.DescriptorImpl.class).setShell(Executables.getPathToExecutable("ash"));
+        project = j.createProject(WorkflowJob.class);
+        credentialsId = UUID.randomUUID().toString();
+
+        project.setDefinition(new CpsFlowDefinition(
+                "node {\n" +
+                        "  withCredentials([string(credentialsId: '" + credentialsId + "', variable: 'CREDENTIALS')]) {\n" +
+                        "    sh '''\n" +
+                        "       echo begin0 $CREDENTIALS end0\n" +
+                        // begin2 => 2 for double quotes
+                        "       echo \"begin2 $CREDENTIALS end2\"\n" +
+                        "    '''\n" +
+                        "  }\n" +
+                        "}", true));
+    }
+
+    @Theory
+    public void credentialsAreMaskedInLogs(String credentials) throws Exception {
+        assumeThat(credentials, not(startsWith("****")));
+
+        CredentialsTestUtil.setStringCredentials(j.jenkins, credentialsId, credentials);
+        WorkflowRun run = runProject();
+
+        j.assertLogContains("begin0 **** end0", run);
+        j.assertLogContains("begin2 **** end2", run);
+        j.assertLogNotContains(credentials, run);
+    }
+
+    private WorkflowRun runProject() throws Exception {
+        return j.assertBuildStatusSuccess(project.scheduleBuild2(0));
```

Comment:
```suggestion
        return j.buildAndAssertSuccess(project);
```

which is so short it should just be inlined anyway.